pkg/pillar/zedcloud/deferred.go

```diff
@@ -137,15 +138,15 @@ func (ctx *DeferredContext) handleDeferred(log *base.LogObject, event time.Time,
 			} else if err != nil {
 				log.Functionf("handleDeferred: for %s status %d failed %s",
 					key, rv.Status, err)
-				exit = true
+				exit = !item.ignoreErr
 				// Make sure we pass a non-zero result to the sentHandler.
 				if rv.Status == types.SenderStatusNone {
 					rv.Status = types.SenderStatusFailed
 				}
 			} else if rv.Status != types.SenderStatusNone {
 				log.Functionf("handleDeferred: for %s received unexpected status %d",
 					key, rv.Status)
-				exit = true
+				exit = !item.ignoreErr
```

Comment:
When we ignore the error and do not exit, for consistency shouldn't we also avoid updating rv.Status? Or should the definition be that the sentHandler can see the error, but we do not exitOnError? (If so it might make sense to rename or at least document what item.ignoreErr means).

Reply:
I do not think this is a good idea to pretend no error happened. We say 'there is an error' but continue processing the queue. The intention is not to halt the queue and ignore the error.

Reply:
So the semantics are ignoreButLogError and not completelyIgnoreError. It would make sense to document that e.g., where the ignoreErr field is defined in the struct.

Reply:
Will add a comment for the SetDeferred() function where this parameter is passed.

Reply:
Did you already add that?